pact/ffi/verifier.py

```diff
@@ -0,0 +1,57 @@
+"""Wrapper to pact reference dynamic libraries using FFI."""
+from enum import Enum, unique
+from typing import NamedTuple, List
+
+from pact.ffi.pact_ffi import PactFFI
+
+
+@unique
+class VerifyStatus(Enum):
+    SUCCESS = 0  # Operation succeeded
+    VERIFIER_FAILED = 1  # The verification process failed, see output for errors
+    NULL_POINTER = 2  # A null pointer was received
+    PANIC = 3  # The method panicked
+    INVALID_ARGS = 4  # Invalid arguments were provided to the verification process
+
+
+class VerifyResult(NamedTuple):
+    return_code: VerifyStatus
+    logs: List[str]
+
+
+class Verifier(PactFFI):
+    """A Pact Verifier Wrapper.
+
+    This interfaces with the Rust FFI crate pact_ffi, specifically the
+    `verifier`_ module.
+
+    .. _verifier:
+        https://docs.rs/pact_ffi/0.0.1/pact_ffi/verifier/index.html
+    """
+
+    def verify(self, args=None) -> VerifyResult:
+        """Call verify method."""
+
+        if args:
+            c_args = self.ffi.new('char[]', bytes(args, 'utf-8'))
+        else:
+            c_args = self.ffi.NULL
+
+        # This fails if called a second time after the library has been loaded
```

Comment:
TODO: pull this out before merging, I wanted to try and setup to initialise the library for each call so it is completely independent, but wasn't able to get that to work